### PR TITLE
Add limits for htId country mapping

### PR DESCRIPTION
**Pull request by oleg-n-k**
Add limits for htId country mapping

### CODE REVIEW

File: HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    {
+        const int numberOfAccommodations = 1000;
+        const int numberOfLocalities = 10;
+            
+        var numberOfAccommodationsInLocalities = await _dbContext.Accommodations
+            .Where(c => c.CountryId == countryId && c.IsActive)
+            .GroupBy(g => g.LocalityId)
+            .Select(g => new {LocalityId = g.Key, NumberOfAccommodations = g.Count()})
+            .OrderByDescending(g => g.NumberOfAccommodations)
+            .Take(numberOfLocalities)
```
Comment (AlexeyShok) [inline HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs:30]: Maybe it's better to take all localities?

**Review by AlexeyShok — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            .ToListAsync();
+
+        var totalNumberOfAccommodations = numberOfAccommodationsInLocalities.Select(a => a.NumberOfAccommodations).Sum();
+        if (totalNumberOfAccommodations < numberOfAccommodations)
+            return await _dbContext.Accommodations
```
Comment (mingazhev) [inline HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs:35]: Will be more readable with curly brackets, I think

File: HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                .Select(a => new ProjectionForCountryHtIdMapping
+                {
+                    AccommodationId = a.Id,
+                    CountryId = a.CountryId,
+                    CountryNames = a.Country.Names,
+                    SupplierAccommodationCodes = a.SupplierAccommodationCodes
+                })
+                .ToListAsync();
+        
+        var balancedNumberOfAccommodationsInLocality = BalanceLimits(numberOfAccommodationsInLocalities.Select(n => (n.LocalityId, n.NumberOfAccommodations)).ToList(), numberOfAccommodations/numberOfLocalities);
```
Comment (mingazhev) [inline HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs:47]: Don't we need whitspaces there? `numberOfAccommodations/numberOfLocalities`

**Review by mingazhev — APPROVED** [review]
With small comments

File: HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs (hunk 1 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs:30]: Has been discussed during daily meeting.

File: HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs (hunk 2 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs:35]: fixed

File: HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs (hunk 3 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Services/LocationMappingInfo/CountryHtIdMappingService.cs:47]: fixed

**Pull request merged**